packages/kbn-ftr-common-functional-ui-services/services/remote/poll_for_log_entry.ts

```diff
@@ -8,87 +8,63 @@
-    subscriber.add(
-      stop$.pipe(first()).subscribe(() => {
-        driver
-          .executeScript(
-            `
-              if (window.flushCoverageToLog) {
-                window.flushCoverageToLog();
-              }
+        // down-level inline script errors
+        if (entry.message.includes('Refused to execute inline script')) {
+          entry.level = logging.getLevel('INFO');
+        }
 
-              console.log(${JSON.stringify(FINAL_MSG)})
-            `
-          )
-          .catch((error) => subscriber.error(error));
+        return true;
       })
-    );
-
-    subscriber.add(
-      poll$
-        .pipe(
-          delay(ms),
-
-          mergeMap(async () => await logCtrl.get(type)),
-
-          // filter and flatten list of entries
-          mergeMap((entries) => {
-            const filtered = entries.filter((entry) => {
-              if (entry.message.includes(FINAL_MSG)) {
-                poll$.complete();
-                return false;
-              }
+    ),
 
-              // ignore react devtools
-              if (entry.message.includes('Download the React DevTools')) {
-                return false;
-              }
+    // resubscribe when parent completes with delay by `ms` milliseconds
+    repeat(),
+    delay(ms),
 
-              // down-level inline script errors
-              if (entry.message.includes('Refused to execute inline script')) {
-                entry.level = logging.getLevel('INFO');
-              }
+    catchError((error, resubscribe) => {
+      if (
+        error instanceof NoSuchSessionError || // session is invalid
+        error instanceof NoSuchWindowError || // browser window crashed
+        error?.message.startsWith('ECONNREFUSED') // webdriver server is not responding, often after one of previous errors
+      ) {
+        // WebDriver session is invalid, sending the last log message
+        return Rx.concat([
+          new logging.Entry('SEVERE', `${FINAL_LOG_ENTRY_PREFIX}: ${error.message}`),
+        ]);
+      } else {
+        return Rx.concat(
+          // log error as a log entry
+          [new logging.Entry('SEVERE', `ERROR FETCHING BROWSR LOGS: ${error.message}`)],
 
-              return true;
-            });
-
-            if (!poll$.isStopped) {
-              // schedule next poll
-              poll$.next(undefined);
-            }
-
-            return filtered;
-          }),
-
-          catchError((error, resubscribe) => {
-            return Rx.concat(
-              // log error as a log entry
-              [new logging.Entry('SEVERE', `ERROR FETCHING BROWSR LOGS: ${error.message}`)],
-
-              // pause 10 seconds then resubscribe
-              Rx.of(1).pipe(delay(10 * 1000), mergeMapTo(resubscribe))
-            );
-          })
-        )
-        .subscribe(subscriber)
-    );
-  });
+          // pause 10 seconds then resubscribe
+          Rx.of(1).pipe(
+            delay(10 * 1000),
```

Comment:
Short delay (e.g. 10 ms) will help to reproduce logs duplication in main. With this change you won't see multiple messages like 
```
warn browser[SEVERE] ERROR FETCHING BROWSR LOGS: This driver instance does not have a valid session ID (did you call WebDriver.quit()?) and may no longer be used.
```